tests/applications/coroutine_mixin_test.py

```diff
@@ -0,0 +1,46 @@
+from logging import DEBUG, getLevelName
+
+from pytest import CaptureFixture
+
+from beamlime.applications.interfaces import BeamlimeApplicationInterface
+from beamlime.logging import get_logger as get_bm_logger
+from beamlime.logging.handlers import BeamlimeStreamHandler
+
+
+def test_daemon_del(capsys: CaptureFixture[str]):
+    DYING_MESSAGE = {
+        "paused": None,
+        "stopped": None,
+        "last-word": "Application instance killed.",
+    }
+
+    class DummyDaemonApp(BeamlimeApplicationInterface):
+        def parse_config(self, _: dict) -> None:
+            ...
+
+        async def _run(self) -> None:
+            ...
+
+        def __del__(self):
+            super().__del__()
+            DYING_MESSAGE["started"] = self._started
+            DYING_MESSAGE["paused"] = self._paused
+            self.debug(DYING_MESSAGE["last-word"])
+
+    logger = get_bm_logger()
+    if not logger.handlers:
+        logger.addHandler(BeamlimeStreamHandler())
+    logger.setLevel(DEBUG)
+
+    app_name = "Dummy Daemon Application"
+    app = DummyDaemonApp(config={}, name=app_name)
+    app.start()
+    del app
+
+    log_output = capsys.readouterr()[-1]
+    assert (
+        f"{app_name:15} | {getLevelName(DEBUG):8} | "
```

Comment:
Having tests based on hard-coded string lengths feels quite brittle. In addition, the `:15` and `:8` are implementation details that the test should not have to know about.
I would suggest to make the requirements less hard, such as simply checking the `app_name` and `DEBUG` are there in the output?